lightning/src/ln/channelmanager.rs

```diff
@@ -6213,6 +6220,255 @@ where
+													None => msgs::OutboundOnionPayload::TrampolineEntrypoint {
+														amt_to_forward: *sender_intended_htlc_amt_msat,
+														outgoing_cltv_value: *cltv_expiry_height,
+														multipath_trampoline_data: None,
+														trampoline_packet: onion_packet.clone(),
+													},
+													Some(blinding_point) => msgs::OutboundOnionPayload::BlindedTrampolineEntrypoint {
+														amt_to_forward: *sender_intended_htlc_amt_msat,
+														outgoing_cltv_value: *cltv_expiry_height,
+														multipath_trampoline_data: None,
+														trampoline_packet: onion_packet.clone(),
+														current_path_key: blinding_point,
+													}
+												};
+											}
+											_ => {
+												unreachable!("Last element must always initially be of type Receive.");
+											}
+										}
+									}
+
+									let onion_keys = onion_utils::construct_onion_keys(&self.secp_ctx, &path, &inter_trampoline_session_priv).map_err(|_| {
+										APIError::InvalidRoute { err: "Pubkey along hop was maliciously selected".to_owned() }
+									}).unwrap();
+									let outer_onion_prng_seed = self.entropy_source.get_secure_random_bytes();
+									let onion_packet = onion_utils::construct_onion_packet(onion_payloads, onion_keys, outer_onion_prng_seed, &payment_hash).unwrap();
+
+									(onion_packet, htlc_msat, htlc_cltv)
+								};
+
+								// Forward the HTLC over the most appropriate channel with the corresponding peer,
+								// applying non-strict forwarding.
+								// The channel with the least amount of outbound liquidity will be used to maximize the
+								// probability of being able to successfully forward a subsequent HTLC.
+								let maybe_optimal_channel = peer_state.channel_by_id.values_mut()
+									.filter_map(Channel::as_funded_mut)
+									.filter_map(|chan| {
+										let balances = chan.get_available_balances(&self.fee_estimator);
+										if outer_value_msat <= balances.next_outbound_htlc_limit_msat &&
+											outer_value_msat >= balances.next_outbound_htlc_minimum_msat &&
+											chan.context.is_usable() {
+											Some((chan, balances))
+										} else {
+											None
+										}
+									})
+									.min_by_key(|(_, balances)| balances.next_outbound_htlc_limit_msat).map(|(c, _)| c);
+								let optimal_channel = match maybe_optimal_channel {
+									Some(chan) => chan,
+									None => {
+										// Fall back to the specified channel to return an appropriate error.
+										if let Some(chan) = peer_state.channel_by_id
+											.get_mut(&forward_chan_id)
+											.and_then(Channel::as_funded_mut)
+										{
+											chan
+										} else {
+											push_trampoline_forwarding_failure(format!("Could not to route to next Trampoline hop {next_node_id} via forwarding channel {outgoing_scid}"), htlc_source, Some(outgoing_scid), 0x2000 | 25, Vec::new());
+											continue;
+										}
+									}
```

Comment:
Is it possible to call `self.can_forward_htlc` here, now that we have the outgoing scid?